app/src/main/java/com/pvp/app/service/FriendServiceImpl.kt

```diff
@@ -67,28 +67,33 @@ class FriendServiceImpl @Inject constructor(
             .await()
             .documents
             .forEach { userDoc ->
-                val user = userDoc.toObject(FriendObject::class.java)
-
-                if (user != null) {
-                    val updatedFriends = user.friends - email
-                    val updatedSentRequests = user.sentRequests - email
-                    val updatedReceivedRequests = user.receivedRequests - email
-
-                    val updatedUser = FriendObject(
-                        friends = updatedFriends,
-                        sentRequests = updatedSentRequests,
-                        receivedRequests = updatedReceivedRequests
-                    )
-
-                    merge(
-                        updatedUser,
-                        userDoc.id
-                    )
+                try {
+                    val user = userDoc.toObject(FriendObject::class.java)
+
+                    if (user != null) {
+                        val updatedFriends = user.friends - email
+                        val updatedSentRequests = user.sentRequests - email
+                        val updatedReceivedRequests = user.receivedRequests - email
+
+                        if (updatedFriends != user.friends || updatedSentRequests != user.sentRequests || updatedReceivedRequests != user.receivedRequests) {
+                            val updatedUser = FriendObject(
+                                friends = updatedFriends,
+                                sentRequests = updatedSentRequests,
+                                receivedRequests = updatedReceivedRequests
+                            )
+
+                            merge(
+                                updatedUser,
+                                userDoc.id
+                            )
+                        }
+                    }
+                } catch (_: Exception) {
```

Comment:
Something is not right here. We shouldn't catch an exception and no nothing with it. Either code should be updated so that the exception is not thrown or something must be done about it (log, throw, ...)